### DISCUSSION TITLE
Config via env vars

### DISCUSSION THREAD

**Discussion by NotAFile** [piqueserver/piqueserver#679] [Feature Requests]
most modern services support some level of configuration via env vars, as they are an easy way to give some isolated context to processes. It would be neat if pique could map config options to env vars somehow.

Env vars to set the config dir and data folder would also be especially useful.

**Comment by feikname** [comment]
See also: https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html

**Comment by NotAFile** [comment]
hm, those are different thing really. You'd want to be able to set the pique data dir independently of the xdg dirs.

**Comment by JohnDowson** [comment]
Not only is that a different thing, it is also irrelevant, as xdg doesn't apply to something that's supposed to be daemonized.

**Comment by NotAFile** [comment]
Yes, I'm not sure how to feel about the ~/.config default myself. It's nice for installing piqueserver on a desktop computer, but a bad default for anything else.
